### PR TITLE
ip leakage for lightningd in non-docker setup fixed

### LINKED ISSUE #1: Second run of setupv2.sh needed to not leak the ISP_IP
Testrun on RaspiBlitz aarch64 v1.8.0rc2

Error:
```
Checking wireguard installation...
Installing wireguard...
> wireguard installed

Error: ipv4: FIB table does not exist.
Flush terminated
Copy wireguard conf file to /etc/wireguard and apply network rules...
```

Full logs:
redacted the ISP_IP TUNNELSATS_VPN_IP VPN_PORT

```
admin@192.168.2.175:~/download/tunnelsats ₿ sudo bash setupv2.sh

###############################
         TunnelSats v2        
         Setup Script         
         Version:             
         v0.0.5
###############################

Looking for WireGuard config file...
> tunnelsatsv2.conf found, proceeding.

RaspiBlitz detected, lnd conf safety check removed

Checking and installing requirements...
Updating the package repositories...

Checking cgroup-tools...
Installing cgroup-tools...
> cgroup-tools installed

Checking nftables installation...
> nftables found

Checking wireguard installation...
Installing wireguard...
> wireguard installed

Error: ipv4: FIB table does not exist.
Flush terminated
Copy wireguard conf file to /etc/wireguard and apply network rules...
> tunnelsatsv2.conf copied to /etc/wireguard/
> network rules applied

Creating cgroup tunnelsats-create-cgroup.sh file in /etc/wireguard/...
> /etc/wireguard/tunnelsats-create-cgroup.sh created.

> tunnelsats-create-cgroup.sh created, executing...
> Successfully added cgroup net_cls subsystem
> Successfully added Mark for net_cls subsystem
> Created tunnelsats cgroup successfully

Creating cgroup systemd service...
Created symlink /etc/systemd/system/multi-user.target.wants/tunnelsats-create-cgroup.service → /etc/systemd/system/tunnelsats-create-cgroup.service.
> tunnelsats-create-cgroup.service: systemd service enabled and started

Creating tunnelsats-splitting-processes.sh file in /etc/wireguard/...
> /etc/wireguard/tunnelsats-splitting-processes.sh created
> tunnelsats-splitting-processes.sh created, executing...
> 2 Process(es) successfully excluded
> tunnelsats-splitting-processes.sh successfully executed

Creating tunnelsats-splitting-processes systemd service...
> tunnelsats-splitting-processes.service created
> tunnelsats-splitting-processes.timer created
Created symlink /etc/systemd/system/multi-user.target.wants/tunnelsats-splitting-processes.service → /etc/systemd/system/tunnelsats-splitting-processes.service.
> tunnelsats-splitting-processes.servicee: systemd service enabled and started
Created symlink /etc/systemd/system/timers.target.wants/tunnelsats-splitting-processes.timer → /etc/systemd/system/tunnelsats-splitting-processes.timer.
> tunnelsats-splitting-processes.timer: systemd timer enabled and started

> lnd.service updated now starts in cgroup tunnelsats

> backup saved under /etc/systemd/system/lnd.service.bak

> lnd.service restarting ...

> lnd.service restarted

Initializing the service...
Created symlink /etc/systemd/system/multi-user.target.wants/wg-quick@tunnelsatsv2.service → /lib/systemd/system/wg-quick@.service.
> wireguard systemd service enabled and started

Verifying tunnel ...
> Tunnel is active
      Your ISP external IP: ISP_IP 
      Your Tunnelsats external IP: TUNNELSATS_VPN_IP

Checking for firewalls and adjusting settings if applicable...
> ufw detected. VPN port rule added

______________________________________________________________________

These are your personal VPN credentials for your lightning configuration.

LND:
#########################################
[Application Options]
listen=0.0.0.0:9735
externalip=TUNNELSATS_VPN_IP:VPN_PORT
[Tor]
tor.streamisolation=false
tor.skip-proxy-for-clearnet-targets=true
#########################################

CLN:
###############################################################################
Umbrel 0.5+
(edit /home/umbrel/umbrel/app-data/core-lightning/docker-compose.yml file 
in section 'lightningd' - 'command' as follows):
comment out the following line: 
#- --bind-addr=:9735
add the following lines:
- --bind-addr=0.0.0.0:9735
- --announce-addr=TUNNELSATS_VPN_IP:VPN_PORT
- --always-use-proxy=false

Native CLN (config file):
bind-addr=0.0.0.0:9735
announce-addr=TUNNELSATS_VPN_IP:VPN_PORT
always-use-proxy=false
###############################################################################

Please save them in a file or write them down for later use.

A more detailed guide is available at: https://blckbx.github.io/tunnelsats/
Afterwards please restart LND / CLN for changes to take effect.

VPN setup completed!
```

The was no error on the second run and the CLN logs show the change:
```
INFO    PEER_1-channeld-chan#77: Peer connection lost
INFO    PEER_1-chan#77: Peer transient failure in CHANNELD_NORMAL: channeld: Owning subdaemon channeld died (62208)
INFO    PEER_1-chan#77: Peer transient failure in CHANNELD_NORMAL: Reconnected
INFO    PEER_2-channeld-chan#78: Peer connection lost
INFO    PEER_2-chan#78: Peer transient failure in CHANNELD_NORMAL: channeld: Owning subdaemon channeld died (62208)
INFO    lightningd: Peer says it sees our address as: ISP_IP.132:53888
INFO    PEER_3-channeld-chan#79: Peer connection lost
INFO    PEER_3-chan#79: Peer transient failure in CHANNELD_NORMAL: channeld: Owning subdaemon channeld died (62208)
INFO    lightningd: Peer says it sees our address as: ISP_IP.132:40076
INFO    PEER_1-channeld-chan#77: Peer connection lost
INFO    PEER_1-chan#77: Peer transient failure in CHANNELD_NORMAL: channeld: Owning subdaemon channeld died (62208)
INFO    PEER_2-channeld-chan#78: Peer connection lost
INFO    PEER_2-chan#78: Peer transient failure in CHANNELD_NORMAL: channeld: Owning subdaemon channeld died (62208)
INFO    lightningd: Peer says it sees our address as: TUNNELSATS_VPN_IP:53890
INFO    PEER_3-channeld-chan#79: Peer connection lost
INFO    PEER_3-chan#79: Peer transient failure in CHANNELD_NORMAL: channeld: Owning subdaemon channeld died (62208)
INFO    lightningd: Peer says it sees our address as: TUNNELSATS_VPN_IP:40078
```

Surpisingly:
```
$ curl api.ipify.org
```
still shows the ISP_IP

```
$ nc -zv TUNNELSAT_VPN_IP VPN_PORT
```
port [tcp/*] succeeded!

### PR DESCRIPTION
**Pull request by ziggie1984**
Fixes #1 and improves overall install|uninstall routines for all interfaces

### CODE REVIEW

File: scripts/uninstallv2.sh [hunk 1]
```diff
@@ -405,9 +356,13 @@ fi
 #Flush nftables and enable old nftables.conf
 #Flush table if exist to avoid redundant rules
 if [ $isDocker -eq 1 ]; then
+  if nft list table ip tunnelsatsv2 &> /dev/null; then
+      echo "Deleting ip tunnelsats nftable..."
+      if nft delete table ip tunnelsatsv2 &> /dev/null; then echo "> Done";echo; fi
+  fi
+  #Make sure old legacy table type inet gets also deleted if present
   if nft list table inet tunnelsatsv2 &> /dev/null; then
```
Comment (blckbx) [inline scripts/uninstallv2.sh:367]: inet -> ip

File: scripts/uninstallv2.sh [hunk 2]
```diff
@@ -116,6 +118,7 @@ do
            break;;
 
    cln|CLN* )     # check CLN (RaspiBlitz)
+        lnImplementation="cln"
```
Comment (openoms) [inline scripts/uninstallv2.sh:121]: it is good to restore the `cl.check.sh`, but it is only checking for plugin related breaking configs and not the `always-use-proxy=false`

Please always set `always-use-proxy=false` to `true` in the CLN config on uninstall.

File: scripts/uninstallv2.sh [hunk 3]
```diff
@@ -405,9 +356,13 @@ fi
...
 if [ $isDocker -eq 1 ]; then
+  if nft list table ip tunnelsatsv2 &> /dev/null; then
+      echo "Deleting ip tunnelsats nftable..."
+      if nft delete table ip tunnelsatsv2 &> /dev/null; then echo "> Done";echo; fi
+  fi
+  #Make sure old legacy table type inet gets also deleted if present
   if nft list table inet tunnelsatsv2 &> /dev/null; then
-      echo "Flushing tunnelsats nftable rules..."
-      if nft flush table inet tunnelsatsv2 &> /dev/null; then echo "> Done";echo; fi
+      if nft delete table inet tunnelsatsv2 &> /dev/null; then echo "> Done";echo; fi
```
[the remark this answers is not in the document]
Comment (ziggie1984) [reply scripts/uninstallv2.sh:368]: we are moving from inet to ip, but I wanted to make sure if a user with the old setup downloads the newest uninstall script, that the legacy table will be deleted as well

File: scripts/uninstallv2.sh (hunk 1 above)
Comment (ziggie1984) [reply scripts/uninstallv2.sh:367]: yes inet will be deprecated, I guess you are still on the old setup, with this change here ? With this change here we move to ip bc we are not planning to make our setup ipv6 ready yet and the table type inet makes some problems with old os version for example buster bc. it has no support for the chain type route in the mangle chain

File: scripts/uninstallv2.sh (hunk 2 above)
Comment (blckbx) [reply scripts/uninstallv2.sh:121]: pushed to main:
for safety
- RaspiBlitz' {lnd,cl}.check.sh restore but edit config files to...
- LND: delete all lines containing tor.skip...
- CLN: switch all `always-use-proxy` to true / 1 (if 0)

Comment (openoms) [reply scripts/uninstallv2.sh:121]: fixing a little issue in https://github.com/blckbx/tunnelsats/pull/7
and this branch will need rebase

Comment (openoms) [reply scripts/uninstallv2.sh:117]: still there is an extra `fi` in line 176

Comment (blckbx) [reply scripts/uninstallv2.sh:117]: > still there is an extra `fi` in line 176

I promise not to code in nano again ... thx

**Comment by openoms** [comment]
The setupv2.sh from here works well from here, no leakage on setup now

One note
Found this comment
```
 please make sure the implementation runs on port 9735?
```
confusing as the default Tor port for CLN (9736 on the raspiblitz) has no interference with the clearnet port.

**Comment by openoms** [comment]
> For CLN to work we need to remap ports on setups because we explicitly tunnel port 9735. Therefore we only support one LN implementation at a time. Do you see potential conflicts with this?

There is no problem, you can configure multiple binds.
See my default CLN example:

from the config:
```
# Tor settings
proxy=127.0.0.1:9050
bind-addr=127.0.0.1:9736
addr=statictor:127.0.0.1:9051/torport=9736
#always-use-proxy=true
rpc-file-mode=0660

# Tunnelsats settings
bind-addr=0.0.0.0:9735
announce-addr=VPN_IP:VPN_PORT
always-use-proxy=false
```

From getinfo
```
   "address": [
      {
         "type": "ipv4",
         "address": "VPN_IP",
         "port": VPN_PORT
      },
      {
         "type": "torv3",
         "address": "xxxxxxxxxxxx.onion",
         "port": 9736
      }
   ],
   "binding": [
      {
         "type": "ipv4",
         "address": "127.0.0.1",
         "port": 9736
      },
      {
         "type": "ipv4",
         "address": "0.0.0.0",
         "port": 9735
      }
   ],
   "version": "v0.11.2",
   "blockheight": 743719,
   "network": "bitcoin",
```

**Comment by blckbx** [comment]
> There is no problem, you can configure multiple binds.

Just thinking through all possible cases. If running RaspiBlitz both LND and CLN, could we run into a problem trying to bind both to port 9735?

**Comment by ziggie1984** [comment]
> > > There is no problem, you can configure multiple binds.
> > 
> > 
> > Just thinking through all possible cases. If running RaspiBlitz with both LND and CLN, could we run into a problem trying to bind both to port 9735?
> 
> Yes, for clearnet it would be an issue and also for binding on the localhost to the same for Tor addresses. The Hidden Address port can be the same on different addresses of course. You should say that running both nodes on clearnet with Tunnelsats is not supported until a custom port is implemented.

still undecided whether we will include it in the script, but its mentioned in the FAQ now

**Pull request merged**